Simulator/Recorders/Hdf5Recorder.cpp

```diff
@@ -189,7 +189,15 @@ void Hdf5Recorder::compileHistories()
                } else if (variableInfo.hdf5Datatype_ == PredType::NATIVE_INT) {
                   vector<int> dataBuffer(variableInfo.variableLocation_.getNumElements());
                   for (size_t i = 0; i < variableInfo.variableLocation_.getNumElements(); ++i) {
-                     dataBuffer[i] = get<int>(variableInfo.variableLocation_.getElement(i));
+                     if constexpr (std::is_same_v<
+                                      vertexType,
+                                      std::decay_t<decltype(get<vertexType>(
+                                         variableInfo.variableLocation_.getElement(i)))>>) {
+                        dataBuffer[i] = static_cast<int>(
+                           get<vertexType>(variableInfo.variableLocation_.getElement(i)));
+                     } else {
+                        dataBuffer[i] = get<int>(variableInfo.variableLocation_.getElement(i));
+                     }
```

Comment:
This is some serious C++-foo. You're going to have to add some comments here so mere mortals like me can understand this.